metrics-monitoring/README.adoc

```diff
@@ -0,0 +1,225 @@
++
+--
+When you create your Kafka instance and topics, the *Dashboard* page is initially empty. After you start producing and consuming messages in your services, you can return to this page to view related metrics.
+
+NOTE: In some cases, after you start producing and consuming messages, you might need to wait several minutes for the latest metrics to appear. You might also need to wait until your instance and topics contain enough data for metrics to appear.
+
+--
+
+[id="proc-configuring-metrics-prometheus_{context}"]
+== Configuring metrics monitoring for a Kafka instance in Prometheus
+
+As an alternative to viewing metrics for a Kafka instance in the {product} web console, you can export your metrics to https://prometheus.io/docs/introduction/overview/[Prometheus] and integrate the metrics with your own metrics monitoring platform. {product} provides a `kafkas/{id}/metrics/federate` API endpoint that you can configure as a scrape target for Prometheus to use to collect and store metrics. You can then access the metrics in the https://prometheus.io/docs/visualization/browser/[Prometheus expression browser] or in a data-graphing tool such as https://prometheus.io/docs/visualization/grafana/[Grafana].
+
+This procedure follows the https://github.com/prometheus-operator/prometheus-operator/blob/main/Documentation/additional-scrape-config.md#additional-scrape-configuration[Additional Scrape Configuration] method defined by Prometheus for integrating third-party metrics.
+
+.Prerequisites
+* You have access to a Kafka instance that contains topics in {product}. For more information about access management in {product}, see {base-url}{access-mgmt-url}[_Managing account access in {product-long}_^].
+* You have the generated credentials for your service account that has access to the Kafka instance. To regenerate and re-copy the credentials, use the *Service Accounts* page in the {product} web console to find your service account and update the credentials. For more information about service accounts, see {base-url}{getting-started-url}#proc-creating-service-account_getting-started[_Getting started with {product-long}_^].
+* You or other users have produced and consumed messages in the Kafka topics. For example, to use Kafka scripts to produce and consume messages, see {base-url}{kafka-bin-scripts-url}[_Configuring and connecting Kafka scripts with {product-long}_^].
+* You've installed a Prometheus instance. For installation instructions, see https://prometheus.io/docs/prometheus/latest/getting_started/[Getting Started] in the Prometheus documentation.
+* You've installed the https://kubernetes.io/docs/tasks/tools/#kubectl[kubectl] Kubernetes command-line tool.
+
+.Procedure
+. In your Prometheus instance, create the following `kafka-federate.yaml` file. Replace the values with your own Kafka instance and credential information.
```

Comment:
```suggestion
. Create the following `kafka-federate.yaml` file. Replace the values with your own Kafka instance and credential information.
```

Reply:
But they can't just create that file anywhere, has to be in the prometheus install folder or in the location of the installed instance, right? That's what I've found when I've gone through this. If that's not the case, we'd still need to tell them where to put that new file, otherwise they might think, "Create it where exactly?"

Reply:
That file is only really used to generate the secret in step 2 so you should be able to create it anywhere in your local directory as long as you update the path in the kubectl cmd:

```
kubectl create secret generic additional-scrape-configs --from-file=<path-to-kafka-federate.yaml> --dry-run -o yaml \
kubectl apply -f - -n <namespace>
```

Reply:
This only applies if you are using the prometheus operator as well.

Reply:
@JameelB @pmuir Pete's last comment about it applying only to the operator got me thinking and I spoke with Pete and learned that really the configuration file method is the more preferred and simpler method, and the additional config method is the alternative if needed for the operator scenarios. So I've actually now restructured this prometheus procedure to essentially reverse the preference.

Can you two have another look at the new procedure and see what you think now? **Use the same doc preview link in the PR description above and refresh the page.**

Reply:
Latest update makes sense and looks good to me :)

Reply:
Thanks @JameelB !